### PR TITLE
docs: add issue and pr templates

### PR DESCRIPTION
**Pull request by NiceAesth**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+blank_issues_enabled: false
```
Comment (NiceAesth) [inline .github/ISSUE_TEMPLATE/config.yml:1]: Are you ok with this? Personally I would prefer if we started moving towards discussions for feature requests

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    - Titles should follow [semantic commit message format](https://sparkbox.com/foundry/semantic_commit_messages)
+-->
+
+# Describe your changes
+
+## Related Issues / Projects
+
+## Checklist
+- [ ] The changes pass pre-commit checks
+- [ ] The changes follow coding style
```
Comment (NiceAesth) [inline .github/pull_request_template.md:12]: todo: add checklist for passing unit testing and following community guidelines

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
Comment (cmyui) [reply .github/ISSUE_TEMPLATE/config.yml:1]: does this mean people can't open an issue with no description? (could they use a custom description that doesn't fit our template still?)

Comment (NiceAesth) [reply .github/ISSUE_TEMPLATE/config.yml:1]: yeah, it means people would have to follow the template when opening issues. can remove this if needed

Comment (NiceAesth) [reply .github/ISSUE_TEMPLATE/config.yml:1]: ideally we could just add a new template for other issues which you can type anything in

Comment (NiceAesth) [reply .github/ISSUE_TEMPLATE/config.yml:1]: added misc issue template, hopefully that should be enough

**Comment by NiceAesth** [comment]
https://github.com/NiceAesth/test-templates

**Review by cmyui — APPROVED** [review]

**Comment by cmyui** [comment]
thanks! this is much nicer imo

**Pull request merged**